sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala

```diff
@@ -774,6 +774,114 @@ case class Encode(value: Expression, charset: Expression)
 }
 
 /**
+ * Replace all substrings of str that match regexp with rep.
+ *
+ * NOTE: this expression is not THREAD-SAFE, we have to create a instance for each thread.
+ */
+case class RegExpReplace(subject: Expression, regexp: Expression, rep: Expression)
+  extends Expression with ImplicitCastInputTypes with CodegenFallback {
```

Comment:
Oh, I didn't notice that before, I thought the codegen will just skip this expression and continue codegen recursively. We need to provide a codegen version for all of the non-leaf expressions. Anyway, I will update the code here firstly.